### PR TITLE
pkg/store: Update README with latest changes

### PR DESCRIPTION
**Pull request by abronan**
Quick update to `pkg/store` documentation.

Signed-off-by: Alexandre Beslic <abronan@docker.com>

### CODE REVIEW

**Comment by vieux** [comment]
LGTM ping @aluzzardi

File: pkg/store/README.md [hunk 1]
```diff
@@ -57,23 +62,22 @@ A new **storage backend** should include those calls:
...
+	DeleteTree(prefix string) error
+	AtomicPut(key string, value []byte, previous *KVPair, options *WriteOptions) (bool, *KVPair, error)
+	AtomicDelete(key string, previous *KVPair) (bool, error)
 }
 ```
 
-To be elligible as a **discovery backend** only, a K/V store implementation should at least offer `Get`, `Put`, `WatchRange`, `GetRange`.
+In the case of Swarm and to be elligible as a **discovery backend** only, a K/V store implementation should at least offer `Get`, `Put`, `WatchTree` and `List`.
+
+`Put` should support usage of `ttls` to be able to remove entries in case of a node failure.
```
Comment (aluzzardi) [inline pkg/store/README.md]: `ttl`

File: pkg/store/README.md [hunk 2]
```diff
@@ -57,23 +62,22 @@ A new **storage backend** should include those calls:
...
+	NewLock(key string, options *LockOptions) (Locker, error)
+	List(prefix string) ([]*KVPair, error)
+	DeleteTree(prefix string) error
+	AtomicPut(key string, value []byte, previous *KVPair, options *WriteOptions) (bool, *KVPair, error)
+	AtomicDelete(key string, previous *KVPair) (bool, error)
 }
 ```
 
-To be elligible as a **discovery backend** only, a K/V store implementation should at least offer `Get`, `Put`, `WatchRange`, `GetRange`.
+In the case of Swarm and to be elligible as a **discovery backend** only, a K/V store implementation should at least offer `Get`, `Put`, `WatchTree` and `List`.
```
Comment (aluzzardi) [inline pkg/store/README.md]: `eligible`

**Comment by aluzzardi** [comment]
LGTM

**Comment by abronan** [comment]
@aluzzardi done

**Pull request merged**